src/test/java/itest/RulesPostFormIT.java

```diff
@@ -0,0 +1,173 @@
+
+    static {
+        NULL_RESULT.put("result", null);
+    }
+
+    @BeforeAll
+    static void setup() throws Exception {
+        testRule = MultiMap.caseInsensitiveMultiMap();
+        testRule.add("name", "Test Rule");
+        testRule.add("targetAlias", "es.andrewazor.demo.Main");
+        testRule.add("description", "AutoRulesIT automated rule");
+        testRule.add("eventSpecifier", "template=Continuous,type=TARGET");
+    }
+
+    @Test
+    void testAddRuleThrowsWhenFormAttributesNull() throws Exception {
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+
+        webClient
+                .post("/api/v2/rules")
+                .putHeader(HttpHeaders.CONTENT_TYPE.toString(), HttpMimeType.MULTIPART_FORM.mime())
+                .sendForm(
+                        null,
+                        ar -> {
+                            assertRequestStatus(ar, response);
+                        });
+        ExecutionException ex =
+                Assertions.assertThrows(ExecutionException.class, () -> response.get());
+        MatcherAssert.assertThat(ex.getCause().getMessage(), Matchers.equalTo("Bad Request"));
+    }
+
+    @Test
+    void testAddRuleThrowsWhenRuleNameAlreadyExists() throws Exception {
+        CompletableFuture<JsonObject> response = new CompletableFuture<>();
+
+        webClient
+                .post("/api/v2/rules")
+                .sendForm(
+                        testRule,
+                        ar -> {
+                            if (assertRequestStatus(ar, response)) {
+                                response.complete(ar.result().bodyAsJsonObject());
+                            }
+                        });
+
+        JsonObject expectedResponse =
+                new JsonObject(
+                        Map.of(
+                                "meta",
+                                        Map.of(
+                                                "type",
+                                                HttpMimeType.PLAINTEXT.mime(),
+                                                "status",
+                                                "Created"),
+                                "data", Map.of("result", "Test_Rule")));
+        MatcherAssert.assertThat(response.get(), Matchers.equalTo(expectedResponse));
+
+        CompletableFuture<JsonObject> duplicatePostResponse = new CompletableFuture<>();
+        webClient
+                .post("/api/v2/rules")
+                .sendForm(
+                        testRule,
+                        ar -> {
+                            assertRequestStatus(ar, duplicatePostResponse);
+                        });
+
+        ExecutionException ex =
+                Assertions.assertThrows(
+                        ExecutionException.class, () -> duplicatePostResponse.get());
+        MatcherAssert.assertThat(ex.getCause().getMessage(), Matchers.equalTo("Conflict"));
+
+        // clean up rule before running next test
```

Comment:
Just to be sure, cleanup like this should be in a `finally` block, to ensure that it's at least attempted even if something goes wrong above.